### PR TITLE
wrap in debug qp collision assertion

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-routing/lib/system/router.js [hunk 1]
```diff
@@ -856,7 +856,7 @@ const EmberRouter = EmberObject.extend(Evented, {
     let handlerInfoLength = handlerInfos.length;
     let leafRouteName = handlerInfos[handlerInfoLength - 1].name;
     let cached = this._qpCache[leafRouteName];
-    if (cached) {
+    if (cached !== undefined) {
```
Comment (bekzod) [inline packages/ember-routing/lib/system/router.js]: avoiding boolean coercion

**Review by rwjblue — COMMENTED** [review]
Looks good, can we also avoid allocating the object for `qpsByUrlKey`?

**Review by rwjblue — COMMENTED** [review]
Maybe you didn't push? I don't see a change to how `qpsByUrlKey` is defined/setup...

**Comment by bekzod** [comment]
oops sorry just pushed now, shouldn't have pushed friday evening with beer

**Comment by bekzod** [comment]
oops sorry just pushed, shouldn't have pushed friday evening with 🍺

File: packages/ember-routing/lib/system/router.js [hunk 2]
```diff
@@ -856,14 +856,16 @@ const EmberRouter = EmberObject.extend(Evented, {
...
       return cached;
     }
 
     let shouldCache = true;
-    let qpsByUrlKey = {};
     let map = {};
     let qps = [];
+    if (DEBUG) {
+      var qpsByUrlKey = {};
```
[the remark this answers is not in the document]
Comment (bekzod) [reply packages/ember-routing/lib/system/router.js]: done

**Comment by bekzod** [comment]
needs restart

**Comment by bekzod** [comment]
then needs merge :P

**Comment by bekzod** [comment]
this good to go

**Review by rwjblue — APPROVED** [review]

**Pull request merged**